### PR TITLE
add background image, background size, and background repeat to `data-section`

### PR DESCRIPTION
**Pull request by chocopowwwa**
[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | No
| New feature? | Yes
| Automated tests included? | No
| Related user documentation PR URL | -
| Related developer documentation PR URL |  -
| Issues addressed (#s or URLs) | -
| BC breaks? | -
| Deprecations? | -

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:

[//]: # ( As applicable: )
#### Steps to reproduce the bug:
1. 
2. 

#### Steps to test this PR:
1. Open Email Builder, Select The `data-section` element
2. Then you'll see additional field in `Customize Section` menu to change the background image, background size, and background repeat, input some value to it, then save the email
3. Make sure the document is the same after and before you save it, and make sure when you re-click the `data-section` element the value appear in the input field 

#### List deprecations along with the new alternative:
1. 
2. 

#### List backwards compatibility breaks:
1. 
2.

### CODE REVIEW

**Comment by npracht** [comment]
Label: Ready to test

**Comment by npracht** [comment]
Label: Feature Request

**Comment by npracht** [comment]
Label: Feature Request

File: app/bundles/CoreBundle/Form/Type/BuilderSectionType.php [hunk 1]
```diff
@@ -36,6 +37,66 @@ public function buildForm(FormBuilderInterface $builder, array $options)
...
+            'required'   => false,
+            'attr'       => [
+                'class'           => 'form-control',
+                'value'           => 'none',
+                'data-slot-param' => 'background-image',
+            ],
+        ]);
+
+        // Content - Background Repeat
+        $builder->add('content-background-repeat', ChoiceType::class,
```
Comment (galvani) [inline app/bundles/CoreBundle/Form/Type/BuilderSectionType.php]: You need to specify use statement for Choicefield

File: app/bundles/CoreBundle/Form/Type/BuilderSectionType.php [hunk 2]
```diff
@@ -36,6 +37,66 @@ public function buildForm(FormBuilderInterface $builder, array $options)
...
+            'required'   => false,
+            'attr'       => [
+                'class'           => 'form-control',
+                'data-slot-param' => 'background-repeat',
+                'data-toggle'     => 'background-repeat',
+            ],
+        ]);
+
+        // Content - Background Size Width
+        $builder->add('content-background-size-width', 'text', [
```
Comment (galvani) [inline app/bundles/CoreBundle/Form/Type/BuilderSectionType.php]: data-slot and data-slot-parent seem wrong, anyway the values won't get stored

File: app/bundles/CoreBundle/Form/Type/BuilderSectionType.php [hunk 3]
```diff
@@ -36,6 +37,66 @@ public function buildForm(FormBuilderInterface $builder, array $options)
...
+            'required'   => false,
+            'attr'       => [
+                'class'           => 'form-control',
+                'data-slot-param' => 'background-size',
+                'data-toggle'     => 'background-size',
+            ],
+        ]);
+
+        // Content - Background Size Height
+        $builder->add('content-background-size-height', 'text', [
```
Comment (galvani) [inline app/bundles/CoreBundle/Form/Type/BuilderSectionType.php]: data-slot and data-slot-parent seem wrong, anyway the value won't get stored

**Comment by alanhartless** [comment]
Hi!

All PRs need to be merged April 4th to make it into 2.13. Please address the issues noted here with enough time to test the changes. 

Also, we use four spaces, not two, for indentions. Looks like this PR changed them all from 4 to 2 for builder.js. Please revert that change. 

Thanks!
Alan

File: app/bundles/CoreBundle/Assets/js/4.builder.js [hunk 4]
```diff
@@ -4,10 +4,10 @@
  * @returns {string}
  */
 Mautic.getUrlParameter = function (name) {
-    name = name.replace(/[\[]/, '\\[').replace(/[\]]/, '\\]');
-    var regex = new RegExp('[\\?&]' + name + '=([^&#]*)');
-    var results = regex.exec(location.search);
-    return results === null ? '' : decodeURIComponent(results[1].replace(/\+/g, ' '));
+  name = name.replace(/[\[]/, '\\[').replace(/[\]]/, '\\]');
```
Comment (alanhartless) [inline app/bundles/CoreBundle/Assets/js/4.builder.js]: We use four spaces, not two, for indentions. Looks like this PR changed them all. Let's revert that. Thanks!

**Comment by chocopowwwa** [comment]
should be fixed,

**Comment by chocopowwwa** [comment]
@dbhurley do you have something like eslint for mautic?

**Comment by npracht** [comment]
@chocopowwwa any chance you solve conflicts now? 2.14.0 should be closed very soon.

**Comment by npracht** [comment]
@heathdutton i've solved conflicts, would you test again ?

**Comment by heathdutton** [comment]
@npracht I think we're good in this case, re-running failed travis.

File: app/bundles/CoreBundle/Assets/js/4.builder.js [hunk 5]
```diff
@@ -554,7 +560,8 @@ Mautic.sanitizeHtmlBeforeSave = function(htmlContent) {
     // Remove the slot focus highlight
     htmlContent.find('[data-slot-focus], [data-section-focus]').remove();
 
-    var customHtml = Mautic.domToString(htmlContent);
+    // Reokace url("${URL}") with url('${URL}')
+    var customHtml = Mautic.domToString(htmlContent).replace(/url\(&quot;(.+)&quot;\)/, 'url(\'$1\')');
```
Comment (heathdutton) [inline app/bundles/CoreBundle/Assets/js/4.builder.js]: Changeset 1.

File: app/bundles/CoreBundle/Assets/js/4.builder.js [hunk 6]
```diff
@@ -743,22 +750,64 @@ Mautic.initSectionListeners = function() {
                 sectionForm.find('#builder_section_wrapper-background-color').val(Mautic.rgb2hex(sectionWrapper.css('backgroundColor')));
             }
 
+            // Prefill The Backgrounf Image
```
Comment (heathdutton) [inline app/bundles/CoreBundle/Assets/js/4.builder.js]: Changeset 2

File: app/bundles/CoreBundle/Assets/js/4.builder.js [hunk 7]
```diff
@@ -743,22 +750,64 @@ Mautic.initSectionListeners = function() {
...
 
             // Handle color change events
-            sectionForm.on('keyup paste change touchmove', function(e) {
+            sectionForm.on('keyup paste change touchmove', function (e) {
                 var field = mQuery(e.target);
-                if (section.length && field.attr('id') === 'builder_section_content-background-color') {
-                    Mautic.sectionBackgroundChanged(section, field.val());
-                } else if (field.attr('id') === 'builder_section_wrapper-background-color') {
-                    Mautic.sectionBackgroundChanged(sectionWrapper, field.val());
+
```
Comment (heathdutton) [inline app/bundles/CoreBundle/Assets/js/4.builder.js]: Changeset 3

File: app/bundles/CoreBundle/Assets/js/4.builder.js [hunk 8]
```diff
@@ -859,43 +911,73 @@ Mautic.initSections = function() {
...
         var focusedSlot = mQuery(this).closest('[data-slot]');
         if (focusedSlot.attr('data-slot') == 'text') {
             Mautic.setTextSlotEditorStyle(parent.mQuery('#slot_text_content'), focusedSlot);
         }
     });
 };
 
-Mautic.rgb2hex = function(orig) {
-    var rgb = orig.replace(/\s/g,'').match(/^rgba?\((\d+),(\d+),(\d+)/i);
+Mautic.sectionBackgroundImageChanged = function (element, imageUrl) {
```
Comment (heathdutton) [inline app/bundles/CoreBundle/Assets/js/4.builder.js]: Changeset 4

**Comment by npracht** [comment]
Closing in favor of https://github.com/mautic/mautic/pull/6820 that is up to date and fixes few bugs.
@heathdutton if you can have a look on the new PR :)

**Pull request closed without merging**